### DISCUSSION TITLE
Is there a way to alias locales in lingui.config?

### DISCUSSION THREAD

**Discussion by crazyyi** [lingui/js-lingui#1524] [Q&A]
For example, Chinese locales have multiple regional locales with their own sematic naming format. But that is not very user friendly for URL accesses. And Japanese/Korean locale naming is different from its top level domain.
```
    'ja',
    'ko',
    'zh-HanS',
    'zh-HanT'
```
Is it possible to alias them in Lingui.config? If not, is it possible to change them somewhere using middleware, and how to do that?

**Comment by timofei-iatsenko** [comment]
```ts
//lingui.config
{
  locales: ["en", "br"],
}
```
runtime app
```ts
const supportedLocalesMap = {
  "en-GB": "en",
  "en-US": "en",
  "pt-BR": "pt",
  "pt-PT": "pt",
}

const userLocale = detectLocale(`[...]`); // for example `en-GB`

export async function loadCatalog(locale: string) {
  const { messages } = await import(`@lingui/loader!../locales/${locale}/messages.po`);
  return messages;
}

const catalogLocale = supportedLocalesMap[supportedLocalesMap]

// will use `en-GB` locale for date / number formatting but load mesages from `en` catalog
i18n.load(userLocale, await loadCatalog(catalogLocale))
i18n.activate(userLocale)
```

locales from lingui.config used only for extracting/compilation (just a list of files which would be created / compiled). In runtime app you can do whatever logic for choosing locale or catalog. There is room for creativity
